pkg/cmd/infra/deployer/deployer.go

```diff
@@ -78,7 +78,7 @@ func NewCommandDeployer(name string) *cobra.Command {
 }
 
 // NewDeployer makes a new Deployer from a kube client.
-func NewDeployer(client kclient.Interface) *Deployer {
+func NewDeployer(client *kclient.Client) *Deployer {
```

Comment:
requiring concrete client implementations (even in constructors) hamstrings us from ever building a joint client, or a client wrapper, etc